### ISSUE TITLE
JOSS submission acceptance blockers

### ISSUE THREAD

**Issue by hausen** [sepandhaghighi/pyrgg#2]
* Missing statement of need: the software generates random graphs, but the user can only choose the number of vertices, min and max weight for the edges, and a range for a random number of edges. I would expect a useful random graph generator to provide some more options, like connectivity, min, max and avg degree.

* Example usage: the README.md states that the output _can_ be used as input to other programs, but have the files generated by this program ever been used as an input to some other algorithm? how?

* Tests: no tests provided

* Other:
  - there is a "gpu" tag in paper.md, why? 
  - ideally, large binary files as the ones in the [datasets](https://github.com/sepandhaghighi/pyrgg/tree/master/datasets) dir should not belong in a git repository. See: http://blog.deveo.com/storing-large-binary-files-in-git-repositories/

**Comment by sepandhaghighi** [comment]
Hi @hausen 

Thanks for review ;-)

1- About statement of need  :  this software is a quick&easy random graph generator and generate input file for other graph processing applications and one of it goals is small input. But your are right, with few more options it can be more useful.

2- About example of usage : yes some of graph processing frameworks like medusa  , cusha and ... use this graph format (DIMACS) , and first pyrgg written to generate synthetic inputs for these frameworks benchmarking against native code in a paper. Then released as a stand-alone software  

3- About test : I added [test cases](https://github.com/sepandhaghighi/pyrgg#test) for functions.  

4- I deleted gpu tag from [paper.md](https://github.com/sepandhaghighi/pyrgg/blob/master/paper/paper.md)

5- Thanks for your hint about large files and repo ;-)
I deleted datasets folder and all of its traces from github with BFG Repo-Cleaner and now size of repo is about 6MB (prev about 90MB!!)

6- Software references are in [paper.bib](https://github.com/sepandhaghighi/pyrgg/blob/master/paper/paper.bib)


Best Regards
Sepand Haghighi

**Comment by sepandhaghighi** [comment]
Hi @hausen 

[Appveyor](https://ci.appveyor.com/project/sepandhaghighi/pyrgg) and [TravisCI](https://travis-ci.org/sepandhaghighi/pyrgg) added for automated tests.

Best Regards
Sepand Haghighi

**Comment by hausen** [comment]
hi @sepandhaghighi ,

1) I don't want to sound too harsh, but your program can be summarized as:
```
    given: n (number of vertices), min_num_edges, max_num_edges, min_weight, max_weight
    num_edges = random(min_num_edges, max_num_edges)
    for 1...num_edges
        print edge from random(1,n) to random(1,n) with weight random(min_weight, max_weight)
```
As you said, it is a quick and easy random graph generator. But a statement of need must explain why someone would pick your random graph generator instead of coding their own or using another one with more features. For instance: your code will likely generate graphs with isolated vertices, but for most uses, isolated vertices can (or must) be eliminated.

2) The question is "have the files generated by this program ever been used as an input to some other algorithm? how?" Can you please provide a real-world example of your code being used as an input to those graph processing frameworks? Is it being used in research? Or was it used before?

3) The tests are not comprehensive; generation of files in the DIMACS format is not covered.

**Comment by sepandhaghighi** [comment]
Hi @hausen 

Thanks for  re-review :-)

1- From one sight your opinion is right, this program is not complex but we can see lots of useful packages with small functions and also same package in different programming languages!
As I mentioned before this package is an easy and quick graph generator and the main aspect of pyrgg is programmability that generate synthetic random graphs easily.
I'm a computer architecture researcher and work on machine learning and deep learning algorithms implementation on GPU(mainly  graph processing algorithms), for one of our research in lab we needed a graph generator  to create synthetic graphs in different sizes to benchmark our project, I searched for a package but there is no useful package in python that  generate DIMACS (need in project) format graphs so I write this project and share it as a stand-alone open source package for a researcher like me in the future.
I recently added JSON format and also have plan to develop this package and add ram management - optimum multi thread system to generate giant graphs,Erdős–Rényi model and ...
You talked about isolated vertices, you are right, in most of cases these vertices must be eliminated. But one part of our research is about a algorithm that need these isolated vertices as cases, so I didn't delete this option.

2- This package can be used for any research program that need DIMACS graph format, and answer of your question  is yes, pyrgg used as a generator in 3 GPU/FPGA based graph processing algorithms and these papers will be  published in future.

3- I add more tests for DIMACS and JSON file formats. [Travis](https://travis-ci.org/sepandhaghighi/pyrgg) , [Appveyor](https://ci.appveyor.com/project/sepandhaghighi/pyrgg)


Best Regards
Sepand Haghighi
